biolink2/example.rkt

```diff
@@ -54,32 +54,51 @@
                  (mat 'close)
                  (printf "Finished processing ~s rows\n" count)))))
 
-;; materialize the (concept curie property value) relation if not present
-(unless (directory-exists? (db-path "concept"))
-  (printf "buildling relation: ~s; ~s\n"
-          (db-path fnin.nodeprop) (db-path "concept"))
-  (let/files ((in (db-path fnin.nodeprop))) ()
-    (materialize-dsv-stream
-      in header.nodeprop
-      ;; input columns
-      '(curie property value) buffer-size (db-path "concept")
-      ;; attribute names, types, and internal key name for indexing
-      '(curie property value) '(string string string) 'id
-      ;; primary table columns (without any sorted-columns)
-      '(((curie property value) . ())
-        ;; index table columns (without any sorted-columns)
-        ((value property id)    . ())))))
+;; materialize a relation if not present
+(define (materialize-relation name fnin header fields types)
+  (unless (directory-exists? (db-path name))
+    (printf "buildling relation: ~s; ~s\n"
+            (db-path fnin) (db-path name))
+    (let/files ((in (db-path fnin))) ()
+      (materialize-dsv-stream
+        in header
+        ;; input columns
+        fields buffer-size (db-path name)
+        ;; attribute names, types, and internal key name for indexing
+        fields types 'id
+        ;; primary table columns (without any sorted-columns)
+        `((,fields . ())
+          ;; index table columns (without any sorted-columns)
+          (,(append (cdr fields) (list 'id)) . ()))))))
```

Comment:
This is clever, but probably not going to help enough most of the time.

A slightly more useful hack might reproduce the index in the original `concept` example, such as:
`(append (reverse (cdr fields)) (list 'id))`